pkg/api/types.go

```diff
@@ -307,6 +310,14 @@ type TestStepConfiguration struct {
 	OpenshiftInstallerCustomTestImageClusterTestConfiguration *OpenshiftInstallerCustomTestImageClusterTestConfiguration `json:"openshift_installer_custom_test_image,omitempty"`
 }
 
+type TestStep struct {
+	Name        string               `json:"name,omitempty"`
+	Image       string               `json:"image,omitempty"`
+	Commands    string               `json:"commands,omitempty"`
+	ArtifactDir string               `json:"artifact_dir,omitempty"`
+	Resources   ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Why not use the resource mapping we already have?

Reply:
Because the granularity there is per-test, this allows for different resource requirements for each step. I don't really understand why it is a separate top-level field instead of being a field in `TestStepConfiguration`, but we can extend it if there is a good reason.